### ISSUE TITLE
build failulre with musl libc

### ISSUE THREAD

**Issue by ncopa** [lavv17/lftp#59]
I get this error with musl libc:

```
make[3]: Entering directory '/home/ncopa/aports/main/lftp/src/lftp-4.4.15/lib'
depbase=`echo argmatch.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../libtool --silent  --tag=CC   --mode=compile ccache gcc -DHAVE_CONFIG_H -I.   -Os -fomit-frame-pointer  -Os -fomit-frame-pointer -Wall -MT argmatch.lo -MD -MP -MF $depbase.Tpo -c -o argmatch.lo argmatch.c &&\
mv -f $depbase.Tpo $depbase.Plo
In file included from argmatch.c:27:0:
./stdbool.h:115:70: error: expected ';', identifier or '(' before '_Bool'
 typedef enum { _Bool_must_promote_to_int = -1, false = 0, true = 1 } _Bool;
                                                                      ^
./stdbool.h:115:70: warning: useless type name in empty declaration [enabled by default]
Makefile:2024: recipe for target 'argmatch.lo' failed
make[3]: *** [argmatch.lo] Error 1
make[3]: Leaving directory '/home/ncopa/aports/main/lftp/src/lftp-4.4.15/lib'
```

I got surprised that gnulib pulls in stdbool.h since its provided from musl libc.

I found that commit 0f475b6ee3b6b3560686aa7a07b69ba9a3cc7e3d forces use of g++ when running the gnulib tests, which indeed makes some gnulib tests fail (because bool is a c++ feature the stdbool.h should only work with C and fail with C++)

Later, when gnulib is actually compiled, gcc is used. 

Basically, since gnulib was not tested with same compiler as it is later built with, things fails.
